### DISCUSSION TITLE
Datenübertragung ESP8266 mit HC05 auf ESP32 Micropython

### DISCUSSION THREAD

**Discussion by kalus-pankow** [micropython/micropython#13782] [ESP8266]
Hallo,
vorab, ich bin Anfänger und 75 Jahre alt.
Ich hatte bisher Temperaturdaten von ESP32 zu ESP32 mit dem HC05 übertragen.
Bevor die Frage kommt, warum ich das interne BLE nicht benutze, bin nicht mit klar gekommen. Ich nutze Micropython und Thonny.
Nun wollte ich Daten vom ESP8266 mit dem HC05 auf den ESP32 übertragen, dachte kein Problem, falsch gedacht.
UART am ESP8266 (Sender): 
UART(1, baudrate=9600, bits=8, parity=None, stop=1, rxbuf=15, timeout=0, timeout_char=2)
UART am ESP32 (Sender) bisher läuft problemlos:
UART(2, baudrate=9600, bits=8, parity=None, stop=1, tx=17, rx=16, rts=-1, cts=-1, txbuf=256, rxbuf=256, timeout=0, timeout_char=0)

Hat das etwas txbuf zu tun, die sehen ja etwas anders aus.

Also es kommen überhaupt an, jedenfalls empfange ich keine.

Hat jemand eine Idee!?
Klaus

**Comment by kalus-pankow** [comment]
Hallo, nochmal weil fehlerhaft!
vorab, ich bin Anfänger und 75 Jahre alt.
Ich hatte bisher Temperaturdaten von ESP32 zu ESP32 mit dem HC05 übertragen.
Bevor die Frage kommt, warum ich das interne BLE nicht benutze, bin nicht mit klar gekommen. Ich nutze Micropython und Thonny.
Nun wollte ich Daten vom ESP8266 mit dem HC05 auf den ESP32 übertragen, dachte kein Problem, falsch gedacht.
UART am ESP8266 (Sender): 
UART(1, baudrate=9600, bits=8, parity=None, stop=1, rxbuf=15, timeout=0, timeout_char=2)
UART am ESP32 (Sender) bisher läuft problemlos:
UART(2, baudrate=9600, bits=8, parity=None, stop=1, tx=17, rx=16, rts=-1, cts=-1, txbuf=256, rxbuf=256, timeout=0, timeout_char=0)

Hat das etwas txbuf zu tun, die sehen ja etwas anders aus.

Also es kommen überhaupt nichts an, jedenfalls empfange ich keine Daten
Klaus.

**Comment by robert-hh** [comment]
The ESP8266 does not support BLE, so you cannot use it. It supports WLAN, which works fine between ESP8266 and ESP32 using the ESP-Now  package.
About UART. At the ESP8266, UART1 is restricted. You can only use it for TX at Pin 2. UART 0 is used for REPL. You could detach REPL from UART, but depending on the ESP8266 board there may be an electrical conflict at the RX line, which prevent using it with a HC05 module. So the problems you have are most likely related to the ESP8266 restrictions for UART.
b.t.w.: The values of txbuf/rxbuf doe not matter much. You should at least receive some data.

**Reply by kalus-pankow** [reply]
Hallo, Danke für die Antwort. Vorweg zu BLE, ich wollte das das auch nich mit ESP8266 versuchen, sondern mit dem ESP32.
Meinen Sie wirklich den Pin 2 ? Wie soll ich den Pin ansteuern, dafür ist gar kein Port vorgegeben (siehe Anlage) Ich habe wie oben geschrieben an Port 15 gelegt.
[ESP8266_Pin.pdf](https://github.com/micropython/micropython/files/14461000/ESP8266_Pin.pdf)
Klaus

**Reply by kalus-pankow** [reply]
Hallo, ich sehe gerade  es gibt ja doch einen  Port 2! Veruche ich gleich mal.
Klaus

**Reply by robert-hh** [reply]
Testing it at my Wemos D1 mini, the output of UART1 is at GPIO2, board pin D4,

**Reply by kalus-pankow** [reply]
Hallo, es geht!!! Seltsam, wenn Sie sich meine Belegungschema ansehen ist  der GPIO 2  der Platinen Pin 15, Haupsache es geht, wäre ich nie drauf gekommen wo doch immer der  der Port 15 beschrieben wir.
Nochmals vielen Dank! Sie sin der Größte!
Klaus
